control/pid_longitudinal_controller/src/pid_longitudinal_controller.cpp

```diff
@@ -422,19 +421,27 @@ PidLongitudinalController::ControlData PidLongitudinalController::getControlData
   // current velocity and acceleration
   control_data.current_motion.vel = m_current_kinematic_state.twist.twist.linear.x;
   control_data.current_motion.acc = m_current_accel.accel.accel.linear.x;
+  control_data.interpolated_traj = m_trajectory;
 
   // nearest idx
-  const size_t nearest_idx = motion_utils::findFirstNearestIndexWithSoftConstraints(
-    m_trajectory.points, current_pose, m_ego_nearest_dist_threshold, m_ego_nearest_yaw_threshold);
-  const auto & nearest_point = m_trajectory.points.at(nearest_idx).pose;
+  const auto current_interpolated_pose =
+    calcInterpolatedTrajPointAndSegment(control_data.interpolated_traj, current_pose);
+  const size_t nearest_idx = current_interpolated_pose.second + 1;
+  const auto & nearest_point = current_interpolated_pose.first;
+
+  // Insert the interpolated point
+  control_data.interpolated_traj.points.insert(
```

Comment:
When you add a new point in the trajectory, please ensure there are no duplicated points. Otherwise, some computation throws an exception.